api/python/provisioner/commands/bootstrap_provisioner.py

```diff
@@ -364,18 +364,6 @@ def _prepare_salt_config(self, run_args, ssh_client, profile_paths):  # noqa: E5
         else:
             masters = load_yaml(masters_pillar_path)
 
-        # TODO IMPROVE many hard coded values
```

Comment:
Hope there is no major impact with these changes...

Reply:
These were bugs in provisioner API also and were handled but in bootstrap those were ignored.
For cluster-id: two cluster-id were getting created and enclosure id was not getting updated in grains.
Handled those and tested those changes. working Fine.